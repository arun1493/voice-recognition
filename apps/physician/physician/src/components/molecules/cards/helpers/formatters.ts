import i18n from "i18next";
import _get from "lodash/get";
import { EMPTY_OBJECT } from "@suki-web/suki-business";

import {
  Allergy,
  AllergyReaction,
  CardMedicalHistoryProps,
  CardSurgicalHistoryProps,
  ParagraphCard,
  Problem
} from "../../../../types";
import { CardType } from "../../../../constants";

const formatParagraphCardData = (data: CardMedicalHistoryProps | CardSurgicalHistoryProps | any, cardType: string) => {
  const paragraphData: Array<ParagraphCard> = [];

  switch (cardType) {
    case CardType.ALLERGIES: {
      const { allergies, nkda } = data || EMPTY_OBJECT;

      allergies &&
        allergies.forEach((allergyItem: Allergy) => {
          const { allergen_name, reactions } = allergyItem || EMPTY_OBJECT;
          let allergyReactionContent = "";
          reactions &&
            reactions.forEach((allergyItem: AllergyReaction) => {
              const { reaction_name } = allergyItem;
              if (reaction_name) allergyReactionContent = allergyReactionContent.concat(reaction_name, ", ");
            });

          allergyReactionContent = allergyReactionContent.slice(0, -2); // to remove comma at the end of string

          if (allergen_name)
            paragraphData.push({
              title: allergen_name,
              content: allergyReactionContent
            });
        });

      if (nkda)
        paragraphData.push({
          content: i18n.t("pages.note.cards.errors.allergiesNkda")
        });
      break;
    }
    case CardType.PAST_MEDICAL_HISTORY: {
      const { active_problems } = data || EMPTY_OBJECT;
      active_problems &&
        active_problems.forEach((activeProblem: Problem) => {
          const description = _get(activeProblem, "note");
          const error = _get(activeProblem, "error");
          const title = _get(activeProblem, "diagnosis.icd_description");
          if (!error) paragraphData.push({ title, content: description });
        });
      break;
    }

    case CardType.PAST_SURGICAL_HISTORY: {
      data &&
        data.forEach((surgicalHistoryItem: Problem) => {
          const { description, note, error } = surgicalHistoryItem;
          if (!error) paragraphData.push({ title: description, content: note });
        });
      break;
    }

    default:
      break;
  }
  return paragraphData;
};

export { formatParagraphCardData };
