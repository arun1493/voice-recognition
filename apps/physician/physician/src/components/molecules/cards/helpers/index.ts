export * from "./labels";
export * from "./formatters";
