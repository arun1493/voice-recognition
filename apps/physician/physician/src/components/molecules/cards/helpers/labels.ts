import i18n from "i18next";

import { CardType, VITAL_SIGNS } from "../../../../constants";

const getCardTitle = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.titles.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.titles.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.titles.vitals");
    default:
      return "";
  }
};

const getVitalSign = (sign: string) => {
  switch (sign) {
    case VITAL_SIGNS.BLOOD_PRESSURE:
      return i18n.t("pages.note.cards.vitalSigns.bloodPressure");
    case VITAL_SIGNS.BMI:
      return i18n.t("pages.note.cards.vitalSigns.bmi");
    case VITAL_SIGNS.HEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.height");
    case VITAL_SIGNS.PULSE:
      return i18n.t("pages.note.cards.vitalSigns.pulse");
    case VITAL_SIGNS.WEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.weight");
    case VITAL_SIGNS.HEART_RATE:
      return i18n.t("pages.note.cards.vitalSigns.heartRate");
    case VITAL_SIGNS.RESPIRATORY_RATE:
      return i18n.t("pages.note.cards.vitalSigns.respiratoryRate");
    case VITAL_SIGNS.TEMPERATURE:
      return i18n.t("pages.note.cards.vitalSigns.temperature");
    default:
      return null;
  }
};

const getErrorMessage = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.errors.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.errors.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.errors.vitals");
    default:
      return "";
  }
};

export { getCardTitle, getErrorMessage, getVitalSign };
