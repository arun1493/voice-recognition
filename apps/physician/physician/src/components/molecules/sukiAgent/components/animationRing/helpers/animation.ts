import { RING_ANIMATION_DURATION, RING_BASE_OPACITY, RING_SCALE, SukiAgentState } from "../../../../../../constants";

function isValidSukiAgentStateForSingleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (isAgentActive && agentStatus === SukiAgentState.AWAKE) || agentStatus === SukiAgentState.PROCESSING;
}

function isValidSukiAgentStateForMultipleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (
    isAgentActive &&
    !isValidSukiAgentStateForSingleRingAnimation(agentStatus, isAgentActive) &&
    agentStatus === SukiAgentState.LISTENING
  );
}

function getDictationRingVariants(ringsCount: number) {
  const dictationRingVariants = [];
  for (let i = ringsCount; i > 0; i--) {
    const RING_VARIANTS = {
      oscillate: {
        opacity: RING_BASE_OPACITY - i * 0.2,
        scale: RING_SCALE + i * 0.01,
        transition: {
          yoyo: Infinity,
          duration: RING_ANIMATION_DURATION,
          type: "tween",
          delay: i * 0.1,
          ease: "easeInOut"
        }
      }
    };

    dictationRingVariants.push(RING_VARIANTS);
  }
  return dictationRingVariants;
}

export {
  getDictationRingVariants,
  isValidSukiAgentStateForMultipleRingAnimation,
  isValidSukiAgentStateForSingleRingAnimation
};
