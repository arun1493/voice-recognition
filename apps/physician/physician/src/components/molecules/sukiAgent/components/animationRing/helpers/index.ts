export * from "./ringSize";
export * from "./animation";
