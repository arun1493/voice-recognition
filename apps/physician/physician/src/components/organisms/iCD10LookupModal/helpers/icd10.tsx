import _isEmpty from "lodash/isEmpty";

const formatText = (text: string, highlight: string) => {
  if (_isEmpty(text)) return;
  const parts = text.split(new RegExp(`(${highlight})`, "gi"));
  return parts.map((part) => (part.toLowerCase() === highlight.toLowerCase() ? <strong>{part}</strong> : part));
};

const formatCode = (code: string) => {
  if (code?.length > 3) {
    return code?.substring(0, 3) + "." + code?.substring(3);
  }
  return code;
};

export { formatCode, formatText };
