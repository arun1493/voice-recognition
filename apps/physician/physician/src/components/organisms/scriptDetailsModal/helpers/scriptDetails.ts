import { ScreenSize } from "@suki-web/suki-ui";

const { SM, XS, XXS } = ScreenSize;

const getWidth = (breakPoint: ScreenSize) => {
  switch (breakPoint) {
    case XXS:
      return "100vw";
    case XS:
      return 484;
    case SM:
      return 664;
    default:
      return 782;
  }
};

const getHeight = (breakPoint: ScreenSize) =>
  breakPoint === XS || breakPoint === XXS ? "100vh" : "calc( 100vh - 56px )";

const getMaxHeight = (breakPoint: ScreenSize) => (breakPoint === XS || breakPoint === XXS ? "100vh" : 0);

export { getHeight, getMaxHeight, getWidth };
