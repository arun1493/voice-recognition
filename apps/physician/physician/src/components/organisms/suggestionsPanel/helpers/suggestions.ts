import i18n from "i18next";

import { SuggestionTypes } from "../../../../constants";

const getTextBySuggestion = (suggestion: string) => {
  switch (suggestion) {
    case SuggestionTypes.PL:
      return i18n.t("components.suggestionsPanel.pl");
    case SuggestionTypes.ED:
      return i18n.t("components.suggestionsPanel.ed");
    default:
      return "";
  }
};

export { getTextBySuggestion };
