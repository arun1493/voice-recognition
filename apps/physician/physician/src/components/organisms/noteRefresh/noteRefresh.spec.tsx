import ReactDOM from "react-dom";
import { ReactPortal } from "react";
import { cleanup, RenderResult } from "@testing-library/react";

import { NoteRefresh } from ".";
import { configureI18n } from "../../../helpers";
import { componentSetupWithStore } from "../../../hocs";
import { shouldMatchSnapshot } from "../../../helpers/snapshot.spec";

const setup = componentSetupWithStore(() => <NoteRefresh />);

describe("NoteRefresh", () => {
  let component: RenderResult;

  beforeEach(() => {
    configureI18n();
    component = setup();
    ReactDOM.createPortal = jest.fn<ReactPortal, any>((element, node) => element);
  });

  afterEach(() => {
    cleanup();
  });

  it("should render successfully", () => {
    const { baseElement } = component;
    expect(baseElement).toBeTruthy();
  });

  it("should contain text", () => {
    const { baseElement } = component;
    expect(baseElement.getElementsByTagName("p")[1].innerHTML).toContain(
      "This can take up to 30 seconds. Please edit and submit the note from Suki before making more edits in EMR."
    );
  });

  it("should contain title", () => {
    const { baseElement } = component;
    expect(baseElement.getElementsByTagName("p")[0].innerHTML).toContain("Refreshing from EMR");
  });

  it("should contain loader", () => {
    const { baseElement } = component;
    expect(baseElement.getElementsByTagName("svg")).toBeTruthy();
  });

  it("should match snapshot", () => {
    shouldMatchSnapshot(NoteRefresh);
  });
});
