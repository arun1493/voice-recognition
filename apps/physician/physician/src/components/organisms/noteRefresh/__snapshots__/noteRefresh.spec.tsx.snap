// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteRefresh should match snapshot 1`] = `
Array [
  <div
    className="modalMaskstyled__StyledBackdrop-sc-169llx0-0 fRODrN"
    onClick={[Function]}
  />,
  <div
    className="modalstyled__StyledModal-sc-1u144rj-1 bSSMMm"
    data-cy=""
    height="auto"
  >
    <div
      aria-label="Close"
      className="modalstyled__StyledCloseButton-sc-1u144rj-0 bSHZet modal-close"
      height={56}
    >
      <div
        className="navigationIconstyled__StyledNavigationIcon-lwvkbw-0 imuqWK"
        data-cy=""
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          className="iconstyled__StyledButton-ifk1iv-0 kgKJBV"
          onClick={[Function]}
        >
          <svg
            height={24}
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.70857 6.29289C7.31805 5.90237 6.68488 5.90237 6.29436 6.29289C5.90383 6.68342 5.90383 7.31658 6.29436 7.70711L10.5865 11.9993L6.2944 16.2914C5.90388 16.6819 5.90388 17.3151 6.2944 17.7056C6.68493 18.0961 7.31809 18.0961 7.70862 17.7056L12.0007 13.4135L16.2928 17.7056C16.6834 18.0961 17.3165 18.0961 17.7071 17.7056C18.0976 17.3151 18.0976 16.6819 17.7071 16.2914L13.4149 11.9993L17.7071 7.70711C18.0976 7.31658 18.0976 6.68342 17.7071 6.29289C17.3166 5.90237 16.6834 5.90237 16.2929 6.29289L12.0007 10.5851L7.70857 6.29289Z"
              fill="#999999"
              xmlns="http://www.w3.org/2000/svg"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="modalHeaderstyled__StyledModalHeader-sc-137lyqk-0 hqHWEz"
      height={52}
    >
      <div
        className="modalHeaderstyled__StyledModalTitle-sc-137lyqk-1 bBaEJZ"
      >
        <p
          className="textstyled__StyledText-sc-1vedkxi-0 QRgfL"
          color="inherit"
          cursor="default"
          onClick={[Function]}
          size="xl"
        >
          Refreshing from EMR
        </p>
      </div>
    </div>
    <div
      className="modalContentstyled__StyledModalContent-wawizf-0 UAtsy"
      height="auto"
      onScroll={[Function]}
    >
      <div
        className="noteRefreshstyled__ContentContainer-so459h-0 kuaaVU"
      >
        <div
          className="noteRefreshstyled__MessageContainer-so459h-2 hHIvjk"
        >
          <p
            className="textstyled__StyledText-sc-1vedkxi-0 fDyqSY"
            color="inherit"
            cursor="default"
            onClick={[Function]}
            size="md"
          >
            This can take up to 30 seconds. Please edit and submit the note from Suki before making more edits in EMR.
          </p>
        </div>
        <div
          className="noteRefreshstyled__LoaderContainer-so459h-1 etbHAg"
        >
          <svg
            enableBackground="new 0 0 0 0"
            height={56}
            id="L9"
            opacity="1"
            version="1.1"
            viewBox="0 0 100 100"
            width={56}
            x="0px"
            xmlSpace="preserve"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <path
              d="M73,50c0-12.7-10.3-23-23-23S27,37.3,27,50 M30.9,50c0-10.5,8.5-19.1,19.1-19.1S69.1,39.5,69.1,50"
              fill="#cccccc"
            >
              <animateTransform
                attributeName="transform"
                attributeType="XML"
                dur="1s"
                from="0 50 50"
                repeatCount="indefinite"
                to="360 50 50"
                type="rotate"
              />
            </path>
          </svg>
        </div>
      </div>
    </div>
  </div>,
]
`;
